tests/tasks/setup/kit-cluster/default.yaml

```diff
@@ -81,25 +84,62 @@ spec:
                 args:
                   - --kube-api-qps=$(params.scheduler-qps)
+        etcd:
+          spec:
+            nodeSelector: 
+              node.kubernetes.io/instance-type: $(params.cp-instance-type)
+            containers: 
+            - name: etcd
+              args:
+                - --quota-backend-bytes=8589934592
       EOL
+      aws sts get-caller-identity
+      #Use service account attached to pod
+      # TOKEN=$(cat /var/run/secrets/kubernetes.io/serviceaccount/token)
+      # kubectl config set-credentials tekton --token=$TOKEN
+      # kubectl config set-context --current --user=tekton
+      #validate 
+      kubectl get pods -A
       kubectl create -f $(workspaces.config.path)/cp.yaml
-
-      cat > "$(workspaces.config.path)/dp.yaml" <<EOL
+      nodes=$(params.node_count)
+      #TODO: Assumption is always nodecount is greater than or equal to 1000 for scalability testing purposes.
+      asgs=$((nodes/1000))
+      asg_name=$(params.cluster-name)-nodes
+      for i in $(seq 1 $asgs)
+      do
+      filename="$(workspaces.config.path)/dp$i"."yaml"
+      echo $filename
+      cat > "$(workspaces.config.path)/dp$i"."yaml"<<EOL
       apiVersion: kit.k8s.sh/v1alpha1
       kind: DataPlane
       metadata:
-        name: $(params.cluster-name)-nodes 
+        name: "$asg_name-$i"
       spec: 
         clusterName: $(params.cluster-name) # Desired Cluster Name
-        nodeCount: $(params.node-count)
+        nodeCount: 1000 #Max nodes we can accomodate per ASG in KIT today
+        subnetSelector:
+          kit/hostcluster: "$(params.host-cluster-name)-dataplane"
+        allocationStrategy: "flexible"
+        instanceTypes: #defaulting to nitro-instances for now
+          - m5.2xlarge
+          - m5.4xlarge
+          - m4.2xlarge
+          - m4.4xlarge
       EOL
-
       #Provision a dataplane plane for the guest cluster
-      kubectl create -f $(workspaces.config.path)/dp.yaml
-
+      ls -larth $(workspaces.config.path)
+      kubectl create -f "$(workspaces.config.path)/dp$i"."yaml"
+      done
+      # sleep for kude-admin config secret for kit guest cluster to appear
+      sleep 30
       #Get the admin KUBECONFIG for the guest cluster from the substrate cluster
       kubectl get secret $(params.cluster-name)-kube-admin-config -ojsonpath='{.data.config}' | base64 -d > $(workspaces.config.path)/kubeconfig
+      # wait for NLB APIServer endpoint to be available
+      sleep 500
       #Deploy CNI plugin to the guest cluster for the nodes to be ready.
       #Todo: CNI installation only works for `us-west-2`, yet to decide how to tackle other regions.
       kubectl --kubeconfig=$(workspaces.config.path)/kubeconfig apply -f https://raw.githubusercontent.com/aws/amazon-vpc-cni-k8s/master/config/v1.9/aws-k8s-cni.yaml
-      sleep 300
+      sleep 60
+      kubectl --kubeconfig=$(workspaces.config.path)/kubeconfig get pods -A 
+      # for nodes to become ready 
+      sleep $((nodes/6))
```

Comment:
Not sure why are we sleeping here as this task is about bringing up cluster and we don't check the DP nodes state after that and what the reasoning behind /6?

Reply:
as part of bringing up the cluster we also spin up dataplane nodes/asgs. So, we before we hand it off to further tasks(if any) to do any other operations on this cluster, this will make sure it takes enough time for the nodes to join cluster, so for  every `1000` nodes it's like 2.7 mins roughly. Based on my experience this is safest and reasonable number i could come up with.

Reply:
With the current state of things with etcd and KIT cluster, all of the nodes becoming `ready` no matter how much ever time we wait is not a reality (very slim possibility).

Reply:
then why do we even wait for some nodes to be ready? We should narrow down whats the issue and fix it.

Reply:
https://github.com/awslabs/kubernetes-iteration-toolkit/issues/86